src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,464 @@
+import htsjdk.samtools.reference.ReferenceSequenceFileWalker;
+import htsjdk.samtools.util.*;
+import htsjdk.tribble.util.ParsingUtils;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFFileReader;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Program to collect error metrics on bases stratified in various ways.
+ *
+ * @author  Yossi Farjoun
+ */
+
+ @CommandLineProgramProperties(
+        summary = "Program to collect error metrics on bases stratified in various ways." +
+                "\n" +
+                "Sequencing errors come in different 'flavors', for example some occur during sequencing while " +
+                "others happen during 'library construction'. They can be correlated with various aspect of " +
+                "the sequencing experiment: position in the read, base context, length of insert and so on. " +
+                "" +
+                "This program enables the user to collect two different kinds of error metrics (one which attempts to " +
+                "distinguish between pre- and post- sequencer errors, and on which doesn't) and a collation of 'stratifiers' " +
+                "each of which assigns bases into various bins. To complicate things further, the stratifiers can be used together " +
+                "to generate a composite stratification. " +
+                "" +
+                "For example:\n" +
+                "The BASE_QUALITY stratifier will place bases in bins according to their declared base quality. " +
+                "The READ_ORDINALITY stratifier will place bases in one of two bins depending on whether their read " +
+                "is 'first' or 'second'. One could generate a composite stratifier BASE_QUALITY,READ_ORDINALITY which will " +
+                "do both as the same time. " +
+                "" +
+                "The resulting metric file will be named according to a provided prefix and a suffix which is generated " +
+                " automatically according to the error metric. " +
+                "To save on run-time, the tool collects many metrics in a single pass and there should be hardly any " +
+                "performance loss when collecting many metrics at the same time, which is why the default already includes a " +
+                "lengthy collection of metrics. " +
+                "" +
+                "The tool takes in a VCF for the sample " +
+                "to skip actually polymorphic bases. As a result it assumes that any disagreement from the reference is an error." +
```

Comment:
`actually polymorphic bases` -> `polymorphic sites`.  It doesn't do the "cute" thing and treat the alternate alleles also as reference (not errors), but simply skips those sites.